modules/services/conky.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.conky;
+
+  configArgument = if cfg.configFile != null then
+    "--config ${cfg.configFile}"
+  else
+    "--config ${config.xdg.configFile."conky/conky.conf".source}";
```

Comment:
This will just use the `/nix/store/...` path, I think you might want something like `${config.home.homeDirectory}/${config.xdg.configFile."conky/conky.conf".target}` (not sure if `XDG_CONFIG_DIR` is set in a systemd service, otherwise just use that...).

Reply:
using the `/nix/store/...` pattern  here as it seems a common way for systemd service in `modules/services` compare to the `config.home.homeDirectory` one. 
- https://github.com/search?q=repo%3Anix-community%2Fhome-manager++path%3Amodules%2Fservices+config.xdg.configFile&type=code
- https://github.com/search?q=repo%3Anix-community%2Fhome-manager++path%3Amodules%2Fservices+config.home.homeDirectory&type=code 

I'm not sure if it's in purpose or just  the legacy reason. let me know if we indeed what to switch to the `home.homeDirectory` one

Reply:
update: use `config.xdg.configHome` seems the good way. I will update it to use that 

-  https://github.com/search?q=repo%3Anix-community%2Fhome-manager++path%3Amodules%2Fservices+config.xdg.configHome&type=code

Reply:
Using `config.xdg.configFile."conky/conky.conf".source` is good here since it will cause the file content to change if the configuration changes. This allows automatic systemd unit restarts.